src/shared/userdb.c

```diff
@@ -306,21 +306,8 @@ static int userdb_on_query_reply(
                 if (r < 0)
                         goto finish;
```

Comment:
I noticed that on failure here, the strings in membership_data are still leaked.

Reply:
Good point. I reworked it to match the style of the two other cases.